### DISCUSSION TITLE
Question regarding the frame rate options (seems inconsistent with ffmpeg command-line)

### DISCUSSION THREAD

**Discussion by buggyyang** [python-ffmpegio/python-ffmpegio#50] [Q&A] [answered]
Thanks for this nice project! I have a question regarding the framerate change. For example,
```
import ffmpegio

with ffmpegio.open(
    "test_video.mp4",
    mode="rv",
    r=24,
    pix_fmt="rgb24",
) as videostream, ffmpegio.open(
    "low_fps.mp4",
    "wv",
    rate_in=24,
    format="mp4",
    vcodec="libx264",
    pix_fmt="yuv420p",
    overwrite=True,
) as videoout:
    for frame in videostream:
        videoout.write(frame)
```
I have a test_video.mp4 with 60fps, I'm trying to convert it to 24fps without changing the playback time. This means I need to drop some frames for the new framerate. Surprisingly, I find the above code seems working well. However, if I use `ffmpeg -r 24 -i test_video.mp4 low_fps.mp4`, the video won't drop any frame but the output video will have longer playback time. I assume the `-r` in ffmpegio here performs differently to ffmpeg command-line? I would be appreciated if you have any insight, thanks!

**Comment by SohamTilekar** [comment]
as much as I know the rate in should be the input video FPS in this case 60

**Comment by tikuma-lsuhsc** [comment, marked as answer]
@buggyyang - Glad you found it useful, and I moved this conversation to Discussions as it is not an `ffmpegio` issue.

> I assume the -r in ffmpegio here performs differently to ffmpeg command-line?

Yes and no. Yes, they are different but no because you are invoking the difference. 

When calling FFmpeg, the option order matters. 

With `ffmpeg -r 24 -i test_video.mp4 -`, you're forcing the input to have 24 fps, i.e., play back in slow-mo as you described, because `-r` here is an input stream option. 

If you want to change the framerate, you need to use the output `-r` option: `ffmpeg -i test_video.mp4 -r 24 -`

In `ffmpegio`, if you give an `ffmpeg` option name (minus the leading dash `-`) as your keyword argument then you are defining `ffmpeg` output options. By appending `_in` to the `ffmpeg` option name, your keyword argument turns into an `ffmpeg` input option. Using `ffmpegio.transcode` as an example,

```
ffmpegio.transcode('test_video.mp4','low_fps.mp4',r=24) # equivalent to ffmpeg -i test_video.mp4 -r 24 -
ffmpegio.transcode('test_video.mp4','slow_mo.mp4',r_in=24) # equivalent to ffmpeg -r 24 -i test_video.mp4 -
```

This applies everywhere including `ffmpegio.open`. Notice that your first call `videostream` opens with the new output rate (thus dropping excess frames) while the second call `videoout` with its input rate, which matches the `videostream` framerate.

Hope this makes sense to you. Also, if you're only transcoding, use the `ffmpegio.transcode` as I show above as it's a whole lot faster than the dual-stream `for` loop solution you have currently.
